### PR TITLE
Decompose footer component

### PR DESCRIPTION
**Pull request by PupS1k**

### CODE REVIEW

File: src/components/Footer/ListLinks.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import Link from './Link';
+
+const ListLinks = ({classNameContainer, classNameChild, children, links}) => (
+  <div className={classNameContainer}>
+    {children}
```
Comment (azhalkouski) [inline src/components/Footer/ListLinks.js]: Do we really want to let people pass children besides array of links?

File: src/components/Footer/ListLinks.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    {links.map(link => (
+      <Link key={link.text || link.image} className={classNameChild}>
```
Comment (azhalkouski) [inline src/components/Footer/ListLinks.js]: Key must be unique. React reconciliation will fail if keys aren't unique.

File: src/components/Footer/ListLinks.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        {link.image && <img className={`${classNameChild}--img`} src={link.image} alt={link.text}/>}
```
Comment (azhalkouski) [inline src/components/Footer/ListLinks.js]: `classNameChild`?

Isn't it too flexible? Maybe ListLink should have its styles and be reusable with the same manners and view everywhere?

File: src/components/Footer/style.css [hunk 4]
```diff
@@ -35,20 +35,18 @@ footer{
     margin-bottom: 20px;
     flex: 1;
 }
-.link-advertising{
+.additional-link--img{
```
Comment (azhalkouski) [inline src/components/Footer/style.css]: 1. This should be styles of  ListLinks.
2. I wonder what `--img` means... Why is it separated with `--`? Did you do this with any methodology in mind?

**Comment by azhalkouski** [comment]
Since you're going to make changes to this PR anyways, let's rename `ListLinks` into `LinksList`.

File: src/components/Footer/ListLinks.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import Link from './Link';
+
+const ListLinks = ({classNameContainer, classNameChild, children, links}) => (
```
Comment (azhalkouski) [inline src/components/Footer/ListLinks.js]: Let's change files with jsx to be `script.jsx` and not `script.js`

**Comment by azhalkouski** [comment]
webpack.config.js `filename: 'index_bundle.js` -> `filename: 'bundle.js`

File: src/components/Footer/style.css [hunk 6]
```diff
@@ -35,20 +35,18 @@ footer{
...
+    width: 30px;
+    height: 20px;
+    margin-right: 7px;
+}
+.additional-link{
     display: flex;
     display: -moz-flex;
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Footer/style.css]: install autoprefixer

File: src/components/Footer/ListLinks.js (hunk 1 above)
Comment (PupS1k) [reply src/components/Footer/ListLinks.js]: fixed

File: src/components/Footer/ListLinks.js (hunk 2 above)
Comment (PupS1k) [reply src/components/Footer/ListLinks.js]: fixed

File: src/components/Footer/ListLinks.js (hunk 3 above)
Comment (PupS1k) [reply src/components/Footer/ListLinks.js]: fixed

File: src/components/Footer/style.css (hunk 4 above)
Comment (PupS1k) [reply src/components/Footer/style.css]: fixed

File: src/components/Footer/ListLinks.js (hunk 5 above)
Comment (PupS1k) [reply src/components/Footer/ListLinks.js]: it's not obligatory

File: src/components/Footer/style.css (hunk 4 above)
Comment (azhalkouski) [reply src/components/Footer/style.css]: I don't understand how you fixed it.

Do you use BEM methodology? If yes, then what does this `--img` mean?

From BEM Quick Start `block-name__element-name_modifier-name`.

https://en.bem.info/methodology/quick-start/#modifier

**Pull request merged**